lib/helpers/UserGroupUiHelper.ts

```diff
@@ -0,0 +1,180 @@
+  private readonly movePermissionBtn: Locator;
+  private readonly sortPermissionBtn: Locator;
+  private readonly cultureAndHostnamesPermissionBtn: Locator;
+  private readonly restrictPublicAccessPermissionBtn: Locator;
+  private readonly rollbackPermissionBtn: Locator;
+  private readonly addSectionsBtn: Locator;
+
+  constructor(page: Page) {
+    super(page);
+    this.userGroupsTabBtn = page.getByRole('tab', {name: 'User Groups'});
+    this.userGroupNameTxt = page.getByLabel('Name', {exact: true});
+    this.browseNodePermissionBtn = page.locator('umb-input-user-permission-verb').getByText('Browse Node');
+    this.createDocumentBlueprintPermissionBtn = page.locator('umb-input-user-permission-verb').getByText('Create Document Blueprint');
+    this.deletePermissionBtn = page.locator('umb-input-user-permission-verb').getByText('Delete', {exact: true});
+    this.createPermissionBtn = page.locator('umb-input-user-permission-verb').getByText('Create', {exact: true});
+    this.notificationsPermissionBtn = page.locator('umb-input-user-permission-verb').getByText('Notifications', {exact: true});
+    this.publishPermissionBtn = page.locator('umb-input-user-permission-verb').getByText('Publish', {exact: true});
+    this.setPermissionsPermissionBtn = page.locator('umb-input-user-permission-verb').getByText('Set permissions');
+    this.unpublishPermissionBtn = page.locator('umb-input-user-permission-verb').getByText('Unpublish', {exact: true});
+    this.updatePermissionBtn = page.locator('umb-input-user-permission-verb').getByText('Update');
+    this.copyPermissionBtn = page.locator('umb-input-user-permission-verb').getByText('Copy', {exact: true});
+    this.movePermissionBtn = page.locator('umb-input-user-permission-verb').getByText('Move', {exact: true});
+    this.sortPermissionBtn = page.locator('umb-input-user-permission-verb').getByText('Sort', {exact: true});
+    this.cultureAndHostnamesPermissionBtn = page.locator('umb-input-user-permission-verb').getByText('Culture and Hostnames', {exact: true});
+    this.restrictPublicAccessPermissionBtn = page.locator('umb-input-user-permission-verb').getByText('Restrict Public Access');
+    this.rollbackPermissionBtn = page.locator('umb-input-user-permission-verb').getByText('Rollback');
+    this.addSectionsBtn = page.locator('[label="Sections"]').getByLabel('open', {exact: true});
+  }
+
+  async clickUserGroupsTabButton() {
+    await this.userGroupsTabBtn.click({force: true});
+  }
+
+  async enterUserGroupName(name: string) {
+    await this.userGroupNameTxt.clear();
+    await this.userGroupNameTxt.fill(name);
+  }
+
+  async isUserGroupWithNameVisible(name: string, isVisible = true) {
+    return await expect(this.page.locator('uui-table-row', {hasText: name})).toBeVisible({visible: isVisible});
+  }
+
+  async clickUserGroupWithName(name: string) {
+    await this.page.getByRole('link', {name: name}).click();
+  }
+
+  async clickBrowseNodePermission() {
```

Comment:
I'd like to suggest creating a method named 'clickPermissionByName(permissionName: string)' to cover clicking the permission. In the test, you can define an array of permission names and then use a loop to click each permission and verify that they are enabled in the UI. 
It will be useful when the list of permissions changes. You will only need to modify the input array in the test, without having to modify the testHelper.